Land model compile time error (Different types in pointer assignment)

While building SMS.ne4_ne4.FC5AV1C-L on Compy, I got the following error:
```
/qfs/people/sing201/acme/srccodes/compy/E3SM/components/clm/src/external_models/emi/src/emi_data_types/EMI_DataDimensionMod.F90:207.7:

       this%first => new_dim
       1
Error: Different types in pointer assignment at (1); attempted assignment of CLASS(emi_data_dimension_type) to TYPE(emi_data_dimension_type)
/qfs/people/sing201/acme/srccodes/compy/E3SM/components/clm/src/external_models/emi/src/emi_data_types/EMI_DataDimensionMod.F90:211.7:

       this%last%next => new_dim
       1
Error: Different types in pointer assignment at (1); attempted assignment of CLASS(emi_data_dimension_type) to TYPE(emi_data_dimension_type)
/qfs/people/sing201/acme/srccodes/compy/E3SM/components/clm/src/external_models/emi/src/emi_data_types/EMI_DataDimensionMod.F90:214.4:

    this%last => new_dim
    1
Error: Different types in pointer assignment at (1); attempted assignment of CLASS(emi_data_dimension_type) to TYPE(emi_data_dimension_type)
gmake: *** [EMI_DataDimensionMod.o] Error 1
gmake: *** Waiting for unfinished jobs....

```

--- a/components/clm/src/external_models/emi/src/emi_data_types/EMI_DataDimensionMod.F90
+++ b/components/clm/src/external_models/emi/src/emi_data_types/EMI_DataDimensionMod.F90
@@ -46,8 +46,8 @@ implicit none
 
   type emi_data_dimension_list_type
      integer                                :: num_dims
-     type(emi_data_dimension_type), pointer :: first
-     type(emi_data_dimension_type), pointer :: last
+     class(emi_data_dimension_type), pointer :: first
+     class(emi_data_dimension_type), pointer :: last
   contains
      procedure, public :: Init             => EMID_Dim_List_Init
      procedure, public :: AddDim           => EMID_Dim_List_AddDim

